client/src/scenes/BootScene.ts

```diff
@@ -0,0 +1,53 @@
+import { Scene } from "phaser";
+import { HathoraClient } from "@hathora/client-sdk";
+import { RoomConnection } from "../connection";
+
+const client = new HathoraClient(process.env.APP_ID as string, process.env.COORDINATOR_HOST);
+
+class BootScene extends Scene {
+  constructor() {
+    super("scene-boot");
+  }
+
+  preload() {
+    this.load.image("player", "player.png");
+    this.load.image("bullet", "bullet.png");
+  }
+
+  create() {
+    getToken().then((token) => {
+      getRoomId(token).then((roomId) => {
+        const connection = new RoomConnection(client, token, roomId);
+        connection.connect();
```

Comment:
should we do a `.then()` on connect as well? that way by the time we load the game scene, we'll know that the connection as been established

Reply:
We could also make the callback function `async` and just await `.connect()` to avoid nesting super deep. What do you think?